src/prisma-service.ts

```diff
@@ -0,0 +1,956 @@
+
+  if (!(await global_snapshot_exists(term))) return notFoundResponse();
+
+  const toCursor = (row) => ({
+    global_snapshot_hash_destination_addr: {
+      global_snapshot_hash: row.global_snapshot_hash,
+      destination_addr: row.destination_addr
+    }
+  });
+  const fromCursor = (row) => ({
+    global_snapshot_hash: row.global_snapshot_hash,
+    destination_addr: row.destination_addr
+  });
+
+  return await paginatedResults(
+    event,
+    toCursor,
+    fromCursor,
+    {
+      where: { global_snapshots: { ...extractTermParam(term) } },
+      orderBy: [{ created_at: Prisma.SortOrder.asc }]
+    },
+    prisma.dag_reward_transactions.findMany,
+    rewardsResponse
+  );
+};
+
+export const globalSnapshotTransactions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { term } = event.pathParameters || {};
+    if (!term) return missingParameterResponse('term');
+
+    if (!(await global_snapshot_exists(term))) return notFoundResponse();
+
+    const query = {
+      where: {
+        dag_blocks: { global_snapshots: { ...extractTermParam(term) } }
+      },
+      include: {
+        dag_blocks: {
+          select: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      },
+      orderBy: [{ created_at: Prisma.SortOrder.desc }]
+    };
+
+    return await paginatedResults(
+      event,
+      toCreatedAtOrdinalCursor,
+      fromCreatedAtOrdinalCursor,
+      query,
+      prisma.dag_transactions.findMany,
+      dagTransactionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const block = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { hash } = event.pathParameters || {};
+    if (!hash) return missingParameterResponse('hash');
+
+    const block = await prisma.dag_blocks.findUnique({
+      where: { hash },
+      include: {
+        dag_transactions: { select: { hash: true } },
+        global_snapshots: true,
+        super: { include: { block_parents: true } }
+      }
+    });
+    return respond(block, dagBlockResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+const transactionsQuery = async (
+  where,
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const query = {
+      ...where,
+      include: {
+        dag_blocks: {
+          include: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      },
+      orderBy: [
+        { created_at: Prisma.SortOrder.desc },
+        { ordinal: Prisma.SortOrder.desc }
+      ]
+    };
+
+    const toCursor = (row) => ({
+      ...toCreatedAtOrdinalCursor(row),
+      hash: row.hash
+    });
+
+    const fromCursor = (row) => ({
+      ...fromCreatedAtOrdinalCursor(row),
+      hash: row.hash
+    });
+
+    return await paginatedResults(
```

Comment:
We need to be able to set default limit per endpoint instead of for all endpoints at once. I think we could support 1k for this endpoint to get the behavior close to what we currently have deployed.